tensorflow_privacy/privacy/membership_inference_attack/membership_inference_attack.py

```diff
@@ -172,6 +174,85 @@ def run_attacks(attack_input: AttackInputData,
       privacy_report_metadata=privacy_report_metadata)
 
 
+def _compute_privacy_risk_score(attack_input: AttackInputData,
+                                num_bins: int = 15) -> SingleRiskScoreResult:
+  """compute each individual point's likelihood of being a member (https://arxiv.org/abs/2003.10595)
+  Args:
+    attack_input: input data for compute privacy risk scores
+    num_bins: the number of bins used to compute the training/test histogram; we set the default as 15
+  
+  Returns:
+    privacy risk score results
+  """
+  
+  # If the loss or the entropy is provided, just use it; 
+  # Otherwise, call the function to compute the loss (you can also choose to compute entropy) 
+  if attack_input.loss_train is not None and attack_input.loss_test is not None:
+    train_values, test_values = attack_input.loss_train, attack_input.loss_test
+  elif attack_input.entropy_train is not None and attack_input.entropy_test is not None:
+    train_values, test_values = attack_input.entropy_train, attack_input.entropy_test
+  else:
+    train_values, test_values = attack_input.get_loss_train(), attack_input.get_loss_test()
+  
+  # Compute the histogram in the log scale
+  small_value = 1e-10
+  train_log_values = np.log(np.maximum(train_values, small_value))
+  test_log_values = np.log(np.maximum(test_values, small_value))
+  
+  min_log_value = np.amin(np.concatenate((train_log_values, test_log_values)))
+  max_log_value = np.amax(np.concatenate((train_log_values, test_log_values)))
+  bins_hist = np.linspace(min_log_value, max_log_value, num_bins+1)
```

Comment:
It looks replacement 
np.linspace -> np.logspace, would make log histograms on train_values/test_values (i.e. w/o logs).
Is it correct replacement?
If yes, could you please update the code? (it's always better to have simpler code for understanding and maintenance)

Reply:
Changed the code to np.logspace!